### PR TITLE
Revert "chore(deps): update helm release postgresql to v11"

### LINKED ISSUE #11522: Helm deployment fails due to password authentication failed for user "backend-user"
## Issue

Helm deployment fails as the backend cannot access the DB
Log of the backstage backend pod
```
(node:1) [DEP0148] DeprecationWarning: Use of deprecated folder mapping "./" in the "exports" field module resolution of the package at /app/node_modules/yam
Update this package.json to use a subpath pattern like "./*". 
(Use `node --trace-deprecation ...` to show where the warning was created)
2022-05-16T10:52:03.741Z backstage info Loaded config from app-config.yaml, app-config.development.yaml, env
2022-05-16T10:52:03.754Z backstage info 1 secrets found in the config which will be redacted
2022-05-16T10:52:03.773Z backstage info Created UrlReader predicateMux{readers=azure{host=dev.azure.com,authed=false},bitbucket{host=bitbucket.org,authed=false},github{host=github.com,authed=true},gitlab{host=gitlab.com,authed=false},awsS3{host=amazonaws.com,authed=false},fetch{}
Backend failed to start up, Error: Failed to connect to the database to make sure that 'backstage_plugin_catalog' exists, error: password authentication failed for user "backend-user"
```

and backstage-lighthouse `error: password authentication failed for user "backend-user" `

Log of the backstage-lighthouse pod
```
yarn run v1.22.4
$ node ./cjs/run.js
info: building express app... {"service":"lighthouse-audit-service","timestamp":"2022-05-16T10:53:07.570Z"}
error: password authentication failed for user "backend-user" {"name":"error","length":108,"severity":"FATAL","code":"28P01","file":"auth.c","line":"338","routin
e":"auth_failed","service":"lighthouse-audit-service","timestamp":"2022-05-16T10:53:32.486Z"}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Log of the postgresql DB reports also
```
2022-05-17 05:44:56.187 GMT [675] DETAIL:  Role "backend-user" does not exist.
    Connection matched pg_hba.conf line 1: "host     all             all             0.0.0.0/0               md5"
2022-05-17 05:45:08.962 GMT [690] FATAL:  password authentication failed for user "backend-user"
```

I get the same error if I try to log on using the password of the secret 
```
kubectl -n backstage get secret/backstage-postgresql -o jsonpath='{.data.postgres-password}' | base64 -d
gmPqVycTqv%

kubectl exec -it backstage-postgresql-0 -n backstage -c postgresql -- bash -c "psql -U backend-user --password"
Password: gmPqVycTqv
psql: error: connection to server on socket "/tmp/.s.PGSQL.5432" failed: FATAL:  password authentication failed for user "backend-user"
command terminated with exit code 2
```
## Database and tables created
```
k exec -it backstage-postgresql-0 -n backstage -c postgresql -- bash -c "PGPASSWORD=gmPqVycTqv psql -U postgres -c \"\l+\""
                                                                    List of databases
   Name    |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges   |  Size   | Tablespace |                Description
-----------+----------+----------+-------------+-------------+-----------------------+---------+------------+--------------------------------------------
 postgres  | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 |                       | 8529 kB | pg_default | default administrative connection database
 template0 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +| 8377 kB | pg_default | unmodifiable empty database
           |          |          |             |             | postgres=CTc/postgres |         |            |
 template1 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +| 8377 kB | pg_default | default template for new databases
           |          |          |             |             | postgres=CTc/postgres |         |            |
(3 rows)

and users 

postgres=# \du
                                   List of roles
 Role name |                         Attributes                         | Member of
-----------+------------------------------------------------------------+-----------
 postgres  | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
```
**Remark**: As you can see the user `backend-user` is not listed and this is why we cannot access the DB, tables, ... and been authenticated

## Steps to reproduce
```
kind cluster created using this bash script: https://github.com/snowdrop/k8s-infra/blob/main/kind/kind-reg-ingress.sh
npx @backstage/create-app
Name: backend
DB: postgresql
cd backend
yarn add --cwd packages/app graphql-ws
yarn build
yarn build-image -t backstage:dev

kind load docker-image backstage:dev

git clone https://github.com/backstage/backstage.git
cd contrib/chart/backstage
helm dependency update

DOMAIN_NAME="192.168.1.90.nip.io"
cat <<EOF > cfg.yml
backend:
  image:
    repository: backstage
    tag: dev
postgresql:
  service:
    port: 5432    
appConfig:
  app:
    baseUrl: https://backstage.$DOMAIN_NAME
    title: Backstage
  backend:
    baseUrl: https://backstage.$DOMAIN_NAME
    cors:
      origin: https://backstage.$DOMAIN_NAME
  lighthouse:
    baseUrl: https://backstage.$DOMAIN_NAME/lighthouse-api
  techdocs:
    storageUrl: https://backstage.$DOMAIN_NAME/api/techdocs/static/docs
    requestUrl: https://backstage.$DOMAIN_NAME/api/techdocs
EOF

kubectl create ns backstage
helm install -f cfg.yml -n backstage backstage .

helm uninstall backstage -n backstage
```

### PR DESCRIPTION
**Pull request by martina-if**
## Description

This reverts commit d0b04813ab991e0701495e1015c3b386be14ced9.

I am reverting this automatic upgrade of the chart dependency because it has breaking changes and needs more
work and testing.

closes #11522

#### :heavy_check_mark: Checklist

- [ ] ~A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))~
- [ ] ~Added or updated documentation~
- [ ] ~Tests for new functionality and regression tests for bug fixes~
- [ ] ~Screenshots attached (for UI changes)~
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by martina-if** [comment]
I'm happy to create an issue for upgrading postgres to v11 again if the core team wants.

**Review by jhaals — COMMENTED** [review]
Just DCO missing, can be added with`git rebase HEAD~1 --signoff && git push -f`

**Review by jhaals — APPROVED** [review]
Thanks!

**Pull request merged**
